azure-quantum/azure/quantum/target/quantinuum.py

```diff
@@ -77,13 +80,26 @@ def submit(
             )
         if input_params is None:
             input_params = {}
+
+        num_shots = kwargs.pop("num_shots", None)
+
         if num_shots is not None:
+            warn(
+                "The 'num_shots' will be deprecated. Please use 'shots' instead.",
+                category=DeprecationWarning,
+            )
+            shots = num_shots
```

Comment:
nit: as I mentioned in a comment above, could you please make a change where `shots` always take precedence over `num_shots` if both are specified? Sorry for confusion